dashboard/config/routes.rb

```diff
@@ -327,6 +328,7 @@ def redirect_to_teacher_dashboard
   get '/api/student_progress/:section_id/:student_id', to: 'api#student_progress', as: 'student_progress'
   get '/api/user_progress/:script_name', to: 'api#user_progress', as: 'user_progress'
   get '/api/user_progress/:script_name/:stage_position/:level_position', to: 'api#user_progress_for_stage', as: 'user_progress_for_stage'
+  get '/api/user_progress/:script_name/:stage_position/:level_position/:level', to: 'api#user_progress_for_stage', as: 'user_progress_for_stage_and_level'
```

Comment:
Yep, that was the change in initApp/loadApp.js. If we don't specify `:level`, we end up showing the wrong levelsource to students who attempt one version of a level and then load another.